task_processing/plugins/mesos/execution_framework.py

```diff
@@ -133,28 +131,27 @@ def enqueue_task(self, task_config):
             log.info('Reviving offers because we have tasks to run.')
 
     def build_framework_info(self):
-        framework = mesos_pb2.FrameworkInfo()
-        framework.user = ""  # Have Mesos fill in the current user.
+        framework = Dict()
+        # TODO: why does this need to be root, can it be "mesos plz figure out"
+        framework.user = 'root'
```

Comment:
Yea, this had to be root for the tests to pass, but I'd love for mesos to just launch tasks as whatever user launched them. Sagar mentioned that another option is having the docker socket owned by users, but I think kwa had issues with that.